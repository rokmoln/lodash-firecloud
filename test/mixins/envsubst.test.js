import _ from '../../src';

// -----------------------------------------------------------------------------

describe('envsubst', function() {
  it('should replace $VAR pattern', function() {
    let env = {
      VAR: 'key'
    };
    let texts = [
      '$VAR',
      'prefix$VAR',
      'prefix $VAR',
      '$VARsuffix',
      '$VAR suffix'
    ];
    let results = _.map(texts, function(text) {
      return _.envsubst(text, env);
    });
    expect(results).toMatchSnapshot();
  });

  // eslint-disable-next-line no-template-curly-in-string
  it('should replace ${VAR} pattern', function() {
    let env = {
      VAR: 'key'
    };
    /* eslint-disable no-template-curly-in-string */
    let texts = [
      '${VAR}',
      'prefix${VAR}',
      'prefix ${VAR}',
      '${VAR}suffix',
      '${VAR} suffix'
    ];
    /* eslint-enable no-template-curly-in-string */
    let results = _.map(texts, function(text) {
      return _.envsubst(text, env);
    });
    expect(results).toMatchSnapshot();
  });
});
