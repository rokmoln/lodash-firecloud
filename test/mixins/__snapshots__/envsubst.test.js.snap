// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`envsubst should replace \${VAR} pattern 1`] = `
Array [
  "key",
  "prefixkey",
  "prefix key",
  "keysuffix",
  "key suffix",
]
`;

exports[`envsubst should replace $VAR pattern 1`] = `
Array [
  "key",
  "prefixkey",
  "prefix key",
  "$VARsuffix",
  "key suffix",
]
`;
